### PR TITLE
Refactor/train test split

### PR DESCRIPTION
**Pull request by gsganden**
`group_train_test_split` is useful on its own, so it should be a public function. It does not have anything directly to do with creating symlinks, so it belongs in a different module. I followed scikit-learn in calling that module `model_selection`.

I am not going to add this function to the README because it is not core to the library. I will document it when I create hosted docs for the repo.

Pull Request Checklist
 - [X] Pull request is based on WIP branch
 - [X] Pull request includes a description of the change and the reason behind it
 - [X] `pytest` passes locally
 - [X] `flake8 creevey tests` passes locally
 - [X] CHANGELOG has been updated
 - [X] Version in `_version.py` has been updated
 - [ ] README and docs have been updated (if applicable)

### CODE REVIEW

File: creevey/model_selection.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import pandas as pd
+from sklearn.model_selection import GroupShuffleSplit
+
+
+def group_train_test_split(
+    X: pd.DataFrame,
+    y: pd.Series,
+    groups: pd.Series,
+    test_size: Optional[Union[float, int]],
```
Comment (sugi-chan) [inline creevey/model_selection.py]: thoughts on a default value of like .2 or something of that sort? sklearn currently follows that protocol, but looks like it will switch to having folks define the train split size.

**Comment by sugi-chan** [comment]
tests work and things make sense, nothing major, just the one thought on setting some default value to test size if none is specified.

File: creevey/model_selection.py (hunk 1 above)
Comment (gsganden) [reply creevey/model_selection.py]: Good thought; there is some value in matching the `sklearn` API, and I did match it in other respects.

That being said, I prefer not to give a default because IMO default test sizes don't make a lot of sense. Fixed percentages don't make a lot of sense as data sets get large (e.g. you don't need 1MM items in your test set no matter how many total items you have), and Creevey is designed for biggish data sets. A fixed number makes more sense in some ways, but I don't want to be raising errors whenever someone tries to run the function on a small dataset for debugging purposes.

Comment (gsganden) [reply creevey/model_selection.py]: Maybe I'll note the lack of default in the docstring.

Comment (sugi-chan) [reply creevey/model_selection.py]: that makes sense, I figured it was something along those lines.

Comment (sugi-chan) [reply creevey/model_selection.py]: adding that to the docstring seems like it would be helpful in the event folks were expecting some default value

**Comment by sugi-chan** [comment]
yeah I think as long as it is well noted like you added to the docstring that there is no default it should be ok.
